packages/babel-plugin-transform-classes/test/fixtures/spec/arrow-func-in-params-of-constructor/output.js

```diff
@@ -0,0 +1,21 @@
+var Root = /*#__PURE__*/babelHelpers.createClass(function Root() {
+  "use strict";
+
+  babelHelpers.classCallCheck(this, Root);
+});
+var Foo = /*#__PURE__*/function (_Root) {
+  "use strict";
+
+  babelHelpers.inherits(Foo, _Root);
+  var _super = babelHelpers.createSuper(Foo);
+  function Foo(cb = () => {
+    console.log("this is", this);
```

Comment:
This is still wrong, it should use `_this`. I think we should just throw when transforming derived constructors with `this` in the default parameters, without the params transform.

Reply:
I meant that I don't think there is an easy way to transform that code, so we could just throw an error saying to also enable `@babel/plugin-transform-parameters`.

Reply:
https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAEBKD28Au0DeBfAUJ0kYDFFoBTADyWIDsATGBZNTaaYeSiJAJwFdgl5OACibMWAI2gBeaIICUUgHyNRopAAsAlhAB0YsJwD8usAC85AbhErW7eCGLaQ8AOaCAROq3QtbgDTRPCFlLFSxmeVQraAhuAAdiIWCowO1gCWk0kOgsdCA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=false&presets=env&prettier=false&targets=&version=7.20.6&externalPlugins=&assumptions=%7B%7D

It seems that the original issue has enabled `@babel/plugin-transform-parameters`